lib/jnpr/junos/utils/fs.py

```diff
@@ -264,6 +265,35 @@ def _decode(fs):
         return dict((_name(fs), _decode(fs)) for fs in rsp.xpath('filesystem'))
 
     # -------------------------------------------------------------------------
+    # directory_usage - filesystem directory usage
+    # -------------------------------------------------------------------------
+
+    def directory_usage(self, path="."):
+        """
+        Returns the directory usage, similar to the unix "du" command.
+
+        :returns: approximate directory usage, including subdirectories, in bytes
+        """
+        rsp = self._dev.rpc.get_directory_usage_information(path=path, depth="0")
+
+        used_space = rsp.findtext("directory/used-space")
+
+        if used_space is None:
+            return None
```

Comment:
I'm not sure returning `None` makes sense for a missing `<used-space>` element. Especially if you implement my recommendation on changing the return value to a `dict`. Instead, I suggest raising a `RpcError` exception.